numba/core/typing/arraydecl.py

```diff
@@ -562,6 +562,14 @@ def generic(self, args, kws):
             ret = record.typeof(idx.literal_value)
             assert ret
             return signature(ret, *args)
+        if isinstance(record, types.Record) and isinstance(idx, types.IntegerLiteral):
```

Comment:
```suggestion
        elif isinstance(record, types.Record) and isinstance(idx, types.IntegerLiteral):
```

To clarify that it is either String or Integer?

Reply:
Thanks, in 9d08da6 I've made this an `elif` and factored out the instance check on the `types.Record` to save testing twice!

Reply:
yeah, that's good!